Find a functionally equivalent code:android.text.SpannableStringBuilder.drawTextRun:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * Don't call this yourself -- exists for Canvas to use internally.
 * {@hide}
 */

@@ -3,15 +3,15 @@
     int contextLen = contextEnd - contextStart;
     int len = end - start;
     if (contextEnd <= mGapStart) {
-        c.drawTextRun(mText, start, len, contextStart, contextLen, x, y, flags, p);
+        c.drawTextRun(mText, start, len, contextStart, contextLen, x, y, isRtl, p);
     } else if (contextStart >= mGapStart) {
-        c.drawTextRun(mText, start + mGapLength, len, contextStart + mGapLength, contextLen, x, y, flags, p);
+        c.drawTextRun(mText, start + mGapLength, len, contextStart + mGapLength, contextLen, x, y, isRtl, p);
     } else {
         char[] buf = TextUtils.obtain(contextLen);
         getChars(contextStart, contextEnd, buf, 0);
-        c.drawTextRun(buf, start - contextStart, len, 0, contextLen, x, y, flags, p);
+        c.drawTextRun(buf, start - contextStart, len, 0, contextLen, x, y, isRtl, p);
         TextUtils.recycle(buf);
     }
 }

